x-pack/plugins/security_solution/public/common/hooks/use_invalid_filter_query.test.tsx

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { createMockStore, kibanaMock, mockGlobalState, TestProviders } from '../mock';
+import type { Query } from '@kbn/es-query';
+import { genHash, useInvalidFilterQuery } from './use_invalid_filter_query';
+
+const getStore = () =>
+  createMockStore(
+    {
+      ...mockGlobalState,
+      app: {
+        ...mockGlobalState.app,
+        errors: [],
+      },
+    },
+    undefined,
+    kibanaMock
+  );
+
+const getProps = () => ({
+  id: 'test-id',
+  kqlError: new Error('boom'),
+  query: { query: ': :', language: 'kuery' },
+  startDate: '2017-01-01T00:00:00.000Z',
+  endDate: '2018-01-02T00:00:00.000Z',
+});
+
+describe('useInvalidFilterQuery', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('invokes error toast with error title and error instance without original stack', () => {
+    const store = getStore();
+    const props = getProps();
+    const InvalidFilterComponent = () => {
+      useInvalidFilterQuery(props);
+      return null;
+    };
+    render(
+      <TestProviders store={store} startServices={kibanaMock}>
+        <InvalidFilterComponent />
+      </TestProviders>
+    );
```

Comment:
Instead of creating a proxy function component in every test case to render the hook indirectly, we can also use _renderHook_ from '@testing-library/react-hooks' to render the hook directly:

```suggestion
    renderHook(useInvalidFilterQuery, { initialProps: props, wrapper: getWrapper(store) });
```

We can also extract the wrapper creation into:

```
const getWrapper = (store: Store): React.FC => {
  // eslint-disable-next-line react/display-name
  return ({ children }) => (
    <TestProviders store={store} startServices={kibanaMock}>
      {children}
    </TestProviders>
  );
};
```

That's how we usually do it. 
What do you think?

Reply:
Absolutely agree, thanks.